openadapt/window/_windows.py

```diff
@@ -1,30 +1,72 @@
 from loguru import logger
-import pygetwindow as pgw
-
+import pywinauto
+from pywinauto import Desktop
+import time
+from pprint import pprint
 
 def get_active_window_state():
-    window = pgw.getActiveWindow()
-    if not window:
-        logger.warning(f"{window=}")
-        return None
-    title = window.title
-    geometry = window.box
-    left, top, width, height = geometry
+    active_window = get_active_window()
+    meta = get_active_window_meta(active_window)
+    data = get_window_data(active_window)
     state = {
-        "title": title,
-        "left": left,
-        "top": top,
-        "width": width,
-        "height": height,
+        "title": meta['texts'][0],
+        "left": meta['rectangle'].left,
+        "top": meta['rectangle'].top,
+        "width": meta['rectangle'].width(),
+        "height": meta['rectangle'].height(),
         # TODO: get window state; see:
         # https://github.com/MLDSAI/OpenAdapt/issues/75#issuecomment-1536762953
-        "meta": None,
+        "meta": meta,
         "data": None,
-        "window_id": None,
+        "window_id": meta['control_id'],
     }
     return state
 
+def get_active_window_meta(active_window) :
+    if not active_window:
+        logger.warning(f"{active_window=}")
+        return None
+    logger.info(f"{active_window.get_properties()}=")
+    return active_window.get_properties()
+
+def get_active_element_state(x, y):
+    active_window = get_active_window()
+    active_element = active_window.from_point(x,y)
+    properties = active_element.get_properties()
+    return properties
+
+def get_active_window():
+    app = pywinauto.application.Application(backend="uia").connect(active_only=True)
+    window = app.active()
+    logger.info(f"{window=}")
+    return window
+
+
+def get_window_data(active_window):
+    state = get_descendants_info(active_window)
+    logger.info(f"{state=}")
+    #state = window.dump_window()
+    return state
+
+def get_descendants_info(window):
+    result = []
+    for child in window.descendants():
+        info = child.get_properties()
+        descendants = get_descendants_info(child)  # Recursively get descendants of the current child
```

Comment:
Please make sure we are not getting duplicates

Reply:
Done